### PR TITLE
adding deutsche-rentenversicherung.de

### PR DESCRIPTION
**Pull request by julianladisch**
Germany's social security insurance authority

### CODE REVIEW

File: src/chrome/content/rules/Deutsche-Rentenversicherung.de.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<ruleset name="Deutsche-Rentenversicherung.de">
+	<target host="www.deutsche-rentenversicherung.de" />
+	<target host=    "deutsche-rentenversicherung.de" />
+	<target host="www.deutsche-rentenversicherung-bund.de" />
+	<target host=    "deutsche-rentenversicherung-bund.de" />
+	<target host="www.drv-bund.de" />
+	<target host=    "drv-bund.de" />
+
+	<rule from="^https?://(?:www\.)?(?:deutsche-rentenversicherung\.de|deutsche-rentenversicherung-bund\.de|drv-bund\.de)/"
```
Comment (reedy) [inline src/chrome/content/rules/Deutsche-Rentenversicherung.de.xml]: The \.de is common to all 3 capture groups... You might aswell just put it after

"^https?://(?:www\.)?(?:deutsche-rentenversicherung|deutsche-rentenversicherung-bund|drv-bund)\.de/"

But you can simplify it further..

"^https?://(?:www\.)?(?:deutsche-rentenversicherung(-bund)?|drv-bund)\.de/"

Comment (julianladisch) [inline src/chrome/content/rules/Deutsche-Rentenversicherung.de.xml]: JavaScript's RegExp compiles the `from` pattern.  Therefore no optimization is needed beforehand.  Instead rules files can be written in the most comprehensible way for humans.

Comment (reedy) [inline src/chrome/content/rules/Deutsche-Rentenversicherung.de.xml]: Maybe so, in which case (-bund)? is probably premature optimisation. But removing the triple repeated \.de seems sensible at least. It's still readable.

Computer optimisation requires CPU time too

Comment (julianladisch) [inline src/chrome/content/rules/Deutsche-Rentenversicherung.de.xml]: Building the deterministic finite automaton (DFA) automatically does the optimization, I cannot measure a time difference between the three pattern compilations.

**Comment by jsha** [comment]
Please simplify these rules as:

```
<rule from="^http:"
           to="https:" />
```

**Comment by donced** [comment]
@julianladisch
Enhancement of this PR #2132.

**Pull request merged**